server/test/order.spec.js

```diff
@@ -0,0 +1,281 @@
+import supertest from 'supertest';
+import assert from 'assert';
+import app from '../../app';
+import genId from '../helpers/utils';
+
+const request = supertest(app);
+const path = '/api/v1/orders';
+
+const newOrder = {
+  userId: 'noencytns',
+  orderId: genId(),
+  orderInfo: [
+    {
+      foodId: 'ghsjhdijnd', quantity: 4
+    }
+  ],
+  orderAddress: '5b ishanwo street, ikotun lagos',
+  orderDateTime: '20/09/2018',
+  orderStatus: 'new'
+};
+
+const newOrderWrongUser = {
+  userId: 'hkfkjhfgg',
+  orderId: genId(),
+  orderInfo: [
+    {
+      foodId: 'ghsjhdijnd', quantity: 4
+    }
+  ],
+  orderAddress: '5b ishanwo street, ikotun lagos',
+  orderDateTime: '20/09/2018',
+  orderStatus: 'new'
+};
+
+const whiteSpaceName = {
+  userId: 'noencytns',
+  orderId: 'rloaghtnnlrv',
+  orderInfo: [
+    { foodId: 'hdhdidnid', quantity: 1 }
+  ],
+  orderAddress: ' ',
+  orderDateTime: '20/09/2018',
+  orderStatus: 'default'
+};
+
+const noAddress = {
+  userId: 'noencytns',
+  orderId: 'rloaghtnnlrv',
+  orderInfo: [
+    { foodId: 'hdhdidnid', quantity: 1 }
+  ],
+  orderAdress: '',
+  orderDateTime: '20/09/2018',
+  orderStatus: 'default'
+};
+const shortAddress = {
+  userId: 'noencytns',
+  orderId: 'rloaghtnnlrv',
+  orderInfo: [
+    {
+      foodId: 'hdhdidnid', quantity: 1
+    }
+  ],
+  orderAddress: 'fjfkk',
+  orderDateTime: '20/09/2018',
+  orderStatus: 'default'
+};
+
+const orderInfoString = {
+  userId: 'noencytns',
+  orderId: 'rloaghtnnlrv',
+  orderInfo: 'jkldjdkj',
+  orderAddress: 'jmlfmlfmlfml',
+  orderDateTime: '20/09/2018',
+  orderStatus: 'default'
+};
+
+const newEdit = {
+  orderStatus: 'accepted'
+};
+
+const wrongEdit = {
+  orderStatus: 'ahdokljd'
+};
+
+const wrongEditFormat = [
+  'orderStatus', 'accepted'
+];
+
+describe('Get all Orders', () => {
+  describe('when Admin request for All fast food order', () => {
+    it('should return fast food orders', (done) => {
+      request.get(path).end((req, res) => {
+        assert.equal(res.status, 200);
+        assert.equal(res.body.orders[0].userId, 'voehnksoe');
+        assert.equal(res.body.orders[0].orderId, 'grildprhr');
+        assert.equal(res.body.orders[0].orderDateTime, '20/09/2018');
+        assert.equal(res.body.orders[0].orderStatus, 'default');
+        done();
+      });
+    });
+  });
+});
+
+describe('Get an Order', () => {
+  describe('when Admin request for A fast food order', () => {
+    it('should return the fast food orders', (done) => {
+      request.get(`${path}/rloahtnnlrv`).send().end((req, res) => {
+        assert.equal(res.status, 200);
+        assert.equal(res.body.order.userId, 'noencytns');
+        assert.equal(res.body.order.orderId, 'rloahtnnlrv');
+        assert.equal(res.body.order.orderDateTime, '20/09/2018');
+        assert.equal(res.body.order.orderStatus, 'default');
+        done();
+      });
+    });
+  });
+
+
+  describe('when Admin request for A fast food order that does not exist',
```

Comment:
These describe blocks could be simple test cases. Why have you decided to use describe blocks?

Reply:
I think you can give good descriptions in the individual test cases. The describe block works best in my opinion to group a bunch of related tests into one logical unit.

Reply:
thank you for the feedback. i will implement that now